lib/active_model/serializer/adapter/json_api/link.rb

```diff
@@ -3,30 +3,45 @@ class Serializer
     module Adapter
       class JsonApi
         class Link
-          def initialize(serializer)
+          def initialize(serializer, value)
             @object = serializer.object
             @scope = serializer.scope
+
+            # Use the return value of the block unless it is nil.
+            if value.respond_to?(:call)
+              self._value = instance_eval(&value)
```

Comment:
agreed it should be a naked instance variable here

my general thinking: setters and getters are, all things being equal, a better way to handle accessing instance values, since it protects the code against changes and typos.  However, there's no need to message dispatch when setting the value in an initializer, so a reader-only is fine there.  As to eschewing accessors all together, I'd probably only do that for performance reasons, or when maybe when I'm certain there code won't change often so mis-typing a variable name is unlikely.